src/reducers/usersReducer.js

```diff
@@ -0,0 +1,38 @@
+import initialState from '../store/initialState';
+import { REGISTER_REQUEST, REGISTER_FAILURE, REGISTER_SUCCESS } from '../actions/actionTypes';
+
+const { user } = initialState;
+const userReducer = (state = user, action) => {
+  const { type } = action;
+  switch (type) {
+    case REGISTER_REQUEST: {
+      return {
+        ...state,
+        isLoading: true,
+        fullNameError: [],
+        emailError: []
+      };
```

Comment:
considering the fact that our initial state looks like
``````````
{
  global: {
    isLoggedIn: false,
    isLoading: false
  },
  user: {
  }
}

```````````

wouldn't your new state look like 

`````````````````
{
  global: {
    isLoggedIn: false,
    isLoading: false
  },
  user: {
  }
   isLoading: true,
   fullNameError: [],
   emailError: []
}

`````````````````

I don't think your changes will make it into the right objects. Check it out, I may be wrong.

Reply:
I am only accessing user object in the store. After a successful signup operation, the usersReducer is having this 
```usersReducer: {
    isLoading: false,
    isSuccess: true,
    fullNameError: [],
    emailError: []
  }
```
as its new state.

My new State looks like this
```
{
  global: {
    isLoggedIn: false,
    isLoading: false
  },
  user: {{
    isLoading: false,
    isSuccess: true,
    fullNameError: [],
    emailError: []
  }
}
```